packages/mantine/src/components/button/Button.tsx

```diff
@@ -1,21 +1,53 @@
 import {Button as MantineButton, ButtonProps as MantineButtonProps} from '@mantine/core';
-import {forwardRef} from 'react';
+import {forwardRef, useState} from 'react';
 
 import {createPolymorphicComponent} from '../../utils';
 import {ButtonWithDisabledTooltip, ButtonWithDisabledTooltipProps} from './ButtonWithDisabledTooltip';
 
-export interface ButtonProps extends MantineButtonProps, ButtonWithDisabledTooltipProps {}
+export interface ButtonProps extends MantineButtonProps, ButtonWithDisabledTooltipProps {
+    /* Callback executed on click */
+    onClick?: (e?: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+const usePromiseOnClick = (callback: (event: React.MouseEvent<HTMLButtonElement>) => void) => {
```

Comment:
```suggestion
const useLoadingHandler = (handler: MouseEventHandler<HTMLButtonElement>) => {
```

Reply:
`usePromiseOnClick` => `useLoadingHandler` makes a whole lot of sense :+1: 

`(handler: MouseEventHandler<HTMLButtonElement>)` => `(callback: (event: React.MouseEvent<HTMLButtonElement>) => void)` I aint so sure. It will break the typecheck and we do pass a `onClick` callback. I think callback is more semantically correct than handler here.

Reply:
Ok I missread for the type thing `mouseEvent` => `mouseEventHandler`

I'll use that, it does seem right indeed.